components/inputs/input-text.js

```diff
@@ -167,10 +170,10 @@ class InputText extends RtlMixin(LitElement) {
 		`;
 		if (this.label && !this.labelHidden) {
 			return html`
-				<label>
+				<label for="${this._inputId}">
 					<span class="d2l-input-label">${this.label}</span>
```

Comment:
Now that the label doesn't wrap the input, I believe the `d2l-input-label` class can be applied to the label itself, and the span can go away.